influxql/parser.go

```diff
@@ -2565,6 +2590,40 @@ func FormatDuration(d time.Duration) string {
 	return fmt.Sprintf("%du", d/time.Microsecond)
 }
 
+// FormatHumanReadableDuration prints a pretty string that combines the
+// components of the duration into a single string.
+func FormatHumanReadableDuration(d time.Duration) string {
+	if d == 0 {
+		return "0s"
+	}
+
+	s := ""
+	if v := d / (7 * 24 * time.Hour); v != 0 {
+		s += fmt.Sprintf("%dw", v)
+		d -= (7 * 24 * time.Hour) * v
+	}
+	if v := d / (24 * time.Hour); v != 0 {
+		s += fmt.Sprintf("%dd", v)
+		d -= (24 * time.Hour) * v
+	}
+	if v := d / time.Hour; v != 0 {
+		s += fmt.Sprintf("%dh", v)
+		d -= time.Hour * v
+	}
+	if v := d / time.Minute; v != 0 {
+		s += fmt.Sprintf("%dm", v)
+		d -= time.Minute * v
+	}
+	if v := d / time.Second; v != 0 {
+		s += fmt.Sprintf("%ds", v)
+		d -= time.Second * v
+	}
+	if s == "" && d != 0 {
+		s += fmt.Sprintf("%du", d)
+	}
+	return s
+}
```

Comment:
It doesn't output microseconds correctly, but I can make the function a lot more simple.